### PR TITLE
Add firefly-itx-3588j Device tree on Rk 6.1 rkr1

### PR DESCRIPTION
**Pull request by SeeleVolleri**

### CODE REVIEW

File: include/dt-bindings/pinctrl/rockchip.h [hunk 1]
```diff
@@ -44,4 +44,21 @@
 
 #define RK_FUNC_GPIO	0
 
+#define PCA_IO0_0		0
```
Comment (SeeleVolleri) [inline include/dt-bindings/pinctrl/rockchip.h:47]: This will create lots warnings, but still, works.

**Comment by SeeleVolleri** [comment]
oh sorry I forgot adding cooling fan

File: arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi
Comment (amazingfate) [inline arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi:1]: please don't touch this file, disable `chosen` node in board devicetree.

**Review by amazingfate — CHANGES_REQUESTED** [review]
And please squash the devicetree commits to one.

File: arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi
Comment (SeeleVolleri) [reply arch/arm64/boot/dts/rockchip/rk3588-linux.dtsi:1]: done, leave `rk3588-linux.dtsi` untouched

**Comment by SeeleVolleri** [comment]
done, squashed

**Comment by SeeleVolleri** [comment]
Oh no, forgot sign gpg

**Comment by SeeleVolleri** [comment]
done

File: arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+/ {
+	model = "Firefly ITX-3588J HDMI(Linux)";
+	compatible = "rockchip,rk3588-firefly-itx-3588j", "rockchip,rk3588";
+};
+
+/ {
+	chosen {
+	status = "disabled";
+	};
```
Comment (amazingfate) [inline arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts]: We usually delete chosen like this: https://github.com/armbian/linux-rockchip/blob/rk-6.1-rkr1/arch/arm64/boot/dts/rockchip/rk3588-rock-5b.dts#L27
Did you test it?

Comment (SeeleVolleri) [reply arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts]: I'm currently building, apologize

Comment (SeeleVolleri) [reply arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts]: I tested, it works fine, fan are fine too

**Comment by SeeleVolleri** [comment]
image

File: include/dt-bindings/pinctrl/rockchip.h [hunk 3]
```diff
@@ -44,4 +44,21 @@
...
+#define PCA_IO0_1		1
+#define PCA_IO0_2		2
+#define PCA_IO0_3		3
+#define PCA_IO0_4		4
+#define PCA_IO0_5		5
+#define PCA_IO0_6		6
+#define PCA_IO0_7		7
+#define PCA_IO0_0		8
```
Comment (amazingfate) [inline include/dt-bindings/pinctrl/rockchip.h]: Is this a typo?
```
 [🐳|🔨]   ./scripts/dtc/include-prefixes/dt-bindings/pinctrl/rockchip.h:55: warning: "PCA_IO0_0" redefined
  [🐳|🔨]      55 | #define PCA_IO0_0               8
  [🐳|🔨]         | 
  [🐳|🔨]   ./scripts/dtc/include-prefixes/dt-bindings/pinctrl/rockchip.h:47: note: this is the location of the previous definition
  [🐳|🔨]      47 | #define PCA_IO0_0               0
  [🐳|🔨]         | 
```

**Review by amazingfate — CHANGES_REQUESTED** [review]

File: include/dt-bindings/pinctrl/rockchip.h (hunk 3 above)
Comment (SeeleVolleri) [reply include/dt-bindings/pinctrl/rockchip.h]: yeah...

**Comment by chainsx** [comment]
I suggest integrating the device tree of this device into a single file. 
Copying the device tree from the BSP directly for submission is in my opinion a downward spiral in overall readability and quality.

**Comment by SeeleVolleri** [comment]
I'm cleaning up dtsi files now

**Comment by SeeleVolleri** [comment]
Cleaned up, now building...

**Comment by amazingfate** [comment]
Makefile change is missing.

**Comment by SeeleVolleri** [comment]
Thanks for reminding me that!

**Comment by SeeleVolleri** [comment]
I got a new problem, fixing on that...
```
[🔨]   Error: arch/arm64/boot/dts/rockchip/rk3588-rk806-single.dtsi:12.26-27 syntax error
[🔨]   FATAL ERROR: Unable to parse input tree
[🔨]   make[3]: *** [scripts/Makefile.lib:422: arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dtb] Error 1
[🔨]   make[3]: *** Waiting for unfinished jobs....
```

File: arch/arm64/boot/dts/rockchip/Makefile [hunk 4]
```diff
@@ -277,6 +277,7 @@ dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-evb7-lp4-v11-linux-ipc.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-evb7-v11.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-evb7-v11-linux.dtb
 dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-fxblox-rk1.dtb
+dtb-$(CONFIG_ARCH_ROCKCHIP) += rk3588-firefly-itx-3588j.dtb
```
Comment (amazingfate) [inline arch/arm64/boot/dts/rockchip/Makefile:279]: Please sort with alphabet.

File: arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts [hunk 5]
```diff
@@ -0,0 +1,1575 @@
...
+#include "dt-bindings/display/drm_mipi_dsi.h"
+#include "dt-bindings/display/rockchip_vop.h"
+#include "dt-bindings/gpio/gpio.h"
+#include "dt-bindings/input/rk-input.h"
+#include "dt-bindings/pinctrl/rockchip.h"
+#include "dt-bindings/pwm/pwm.h"
+#include "dt-bindings/sensor-dev.h"
+#include "dt-bindings/usb/pd.h"
+#include "rk3588-rk806-single.dtsi"
+#include "rk3588.dtsi"
```
Comment (amazingfate) [inline arch/arm64/boot/dts/rockchip/rk3588-firefly-itx-3588j.dts]: Please include `rk3588s.dtsi` before `rk3588-rk806-single.dtsi`.

**Review by amazingfate — CHANGES_REQUESTED** [review]

**Comment by SeeleVolleri** [comment]
`make ARCH=arm64 dtbs` has no errors, I'm going to build armbian tomorrow, my UPS is running out of power.
And thanks @chainsx for pointing out build failure because `rockchip.h` missing pins!

**Comment by SeeleVolleri** [comment]
I made a very dumb mistake before: Somehow uboot reads `rk3588s-roc-pc.dtb` and I keep editing `rk3588-firefly-itx3588j`dts to correct the periphels error.
Now it boots flawlessly without any mistake or missing devices, all working!
Apologize for verbose pr and commits.
Thanks!

**Review by amazingfate — APPROVED** [review]

**Review by chainsx — APPROVED** [review]

**Pull request merged**